cart/interfaces/graphql/schema.graphql

```diff
@@ -17,15 +17,98 @@ type Commerce_Cart {
     appliedCouponCodes: [Commerce_CartCouponCode!]
     defaultCurrency: String!
     totalitems: [Commerce_CartTotalitem!]
+    isPaymentSelected: Boolean!
+    appliedGiftCards: [Commerce_CartAppliedGiftCard!]
+    getDeliveryByCode(deliveryCode: String!): Commerce_CartDelivery
+    getDeliveryCodes: [String!]
+
+    getMainShippingEMail: String!
+    isEmpty: Boolean!
+    hasDeliveryForCode(deliveryCode: String!): Boolean!
+
+    getDeliveryByItemID(itemID: String!): Commerce_CartDelivery!
+    getByItemID(itemID: String!): Commerce_CartItem!
+    getTotalQty(marketPlaceCode: String!, variantCode: String!): Int
+    getByExternalReference(ref: String!): Commerce_CartItem!
+
     itemCount: Int!
     productCount: Int!
-    isPaymentSelected: Boolean!
+
+    getVoucherSavings: Commerce_Price!
     grandTotal: Commerce_Price!
+
+    sumTaxes: Commerce_Cart_Taxes!
+    getCartTeaser: Commerce_Cart_Teaser!
+
+    sumShippingNet: Commerce_Price!
+    sumShippingNetWithDiscounts: Commerce_Price!
+    hasShippingCosts: Boolean!
+    allShippingTitles: [String!]
+
+    subTotalGross: Commerce_Price!
     sumTotalTaxAmount: Commerce_Price!
     subTotalNet: Commerce_Price!
-    appliedGiftCards: [Commerce_CartAppliedGiftCard!]
-    getDeliveryByCode(deliveryCode: String!): Commerce_CartDelivery
-    getDeliveryCodes: [String!]
+
+    subTotalGrossWithDiscounts: Commerce_Price!
+    subTotalNetWithDiscounts: Commerce_Price!
+    sumTotalDiscountAmount: Commerce_Price!
+    sumNonItemRelatedDiscountAmount: Commerce_Price!
+    sumItemRelatedDiscountAmount: Commerce_Price!
+    hasAppliedCouponCode: Boolean!
+
+    getPaymentReference: String!
+    getTotalItemsByType(typeCode: String!): [Commerce_CartTotalitem!]
+    grandTotalCharges: Commerce_Price_Charges!
+
+    hasAppliedGiftCards: Boolean!
+    hasRemainingGiftCards: Boolean!
+    sumAppliedGiftCardsForced: Commerce_Price!
```

Comment:
Forced in the shema sound like a not so nice property name?

Reply:
Yup, we could remap this in field in graphql service. What do you think about general convention with the Forced suffix? 

Pattern i've found is methods with multiple returns are wrapped by the ...Forced  methods, like for example ```.ChargesByType().GetByTypeForced(...)```